solutions/usaco-834.mdx

```diff
@@ -0,0 +1,68 @@
+---
+id: usaco-1037
+source: USACO Bronze 2020 US Open
+title: Cowntact Tracing
+author: Brad Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_sort_silver_open18.html)
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N \cdot T^2)$
+
+<LanguageSection>
+
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class OutOfSorts {
+
+	// hold values
+	static class Entry implements Comparable<Entry> {
+		public int value;
+		public int index;
+
+		public Entry (int v, int i) {
+			value = v;
+			index = i;
+		}
+
+		public int compareTo(Entry other) { // bubble sort
```

Comment:
bubble sort? why? it seems like you're just comparing by value as first precedence, and then index

Reply:
^

Reply:
that is how you compare an entry to another to determine which one will go in front. and then calling arrays.sort on class Entry with this compareTo method in the class will make it do the bubble sort algorithm

Reply:
it does not do bubble sort ?

if it did it would TLE since N <= 1e5 and bubble sort is O(n^2)

Reply:
you're right it doesn't actually perform bubble sort. the sorting just makes it easier to compute the number of moos